File: src/OK.ShortLink.Common/Models/VisitorModel.cs
using System;

namespace OK.ShortLink.Common.Models
{
    public class VisitorModel
    {
        public int Id { get; set; }

        public int LinkId { get; set; }

        public LinkModel Link { get; set; }

        public string IPAddress { get; set; }

        public string UserAgent { get; set; }

        public string OSInfo { get; set; }

        public string DeviceInfo { get; set; }

        public string BrowserIndo { get; set; }

        public DateTime CreatedDate { get; set; }
        
        public DateTime UpdatedDate { get; set; }
    }
}

File: src/OK.ShortLink.Web/src/app/_models/visitor.model.ts
import { Link } from './link.model';

export class Visitor {
  constructor(
    public id: number,
    public linkId: number,
    public link: Link,
    public ipAddress: string,
    public userAgent: string,
    public osInfo: string,
    public deviceInfo: string,
    public browserInfo: string,
    public createdDate: Date,
    public updatedDate: Date
  ) {}
}
